unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1950,18 +1950,25 @@ protected void updateAllAgentCollidersForVisibilityCheck(bool enableColliders)
             return Time.time;
         }
 
-        private bool isSimObjVisible(Camera agentCamera, SimObjPhysics sop) {
+        private bool isSimObjVisible(Camera agentCamera, SimObjPhysics sop, bool includeInvisible, float maxDistance = 0.0f) {
             bool visible = false;
             //check against all visibility points, accumulate count. If at least one point is visible, set object to visible
-            if (sop.VisibilityPoints == null || sop.VisibilityPoints.Length > 0) 
+            if (sop.VisibilityPoints != null && sop.VisibilityPoints.Length > 0) 
             {
                 Transform[] visPoints = sop.VisibilityPoints;
                 int visPointCount = 0;
 
+
                 foreach (Transform point in visPoints) 
                 {
+                    if (maxDistance > 0.0f) {
+                        Vector3 agentPoint = new Vector3(this.transform.position.x, point.position.y, this.transform.position.z);
+                        if (Vector3.Distance(agentPoint, point.position) > maxDistance) {
+                            continue;
+                        }
```

Comment:
The type of distance we use for visibility checks actually isn't "sphere shaped" as it, more or less, ignores y positions. This is important so that objects on the ground are considered interactable. I worry that using the euclidean distance here might thus incorrectly throw away some visibility points.

Reply:
See the change above where we fix the y value of the agentPosition to the visibility point's y position.